Autocomplete/recent-searches/src/index.js

```diff
@@ -0,0 +1,45 @@
+import algoliasearch from 'algoliasearch/lite';
+import { autocomplete, highlightHit } from '@algolia/autocomplete-js';
+import { createLocalStorageRecentSearchesPlugin } from '@algolia/autocomplete-plugin-recent-searches';
+
+const recentSearches = createLocalStorageRecentSearchesPlugin({
+  key: 'RECENT_SEARCH',
+  limit: 5,
+});
+
+const searchClient = algoliasearch(
+  'latency',
+  '6be0576ff61c053d5f9a3225e2a90f76'
+);
+const index = searchClient.initIndex('instant_search');
+
+function searchAndShowMainResults(query) {
+  index.search(query).then((result) => {
+    console.log({
+      highlighted: highlightHit({ attribute: 'name', hit: result.hits[0] }),
+      result,
+    });
+    document.querySelector('#hits').innerHTML = result.hits
+      .map(
+        (hit) => `
+        <article class="hit">
+          <div class="image-wrapper"><img src="${hit.image}" /></div>
+          <p class="name">${hit._highlightResult.name.value}</p>
+          <p class="price">$${hit.price}</p>
+        </article>
+      `
+      )
+      .join('');
+  });
+}
+
+autocomplete({
+  container: '#autocomplete',
+  plugins: [recentSearches],
+  openOnFocus: true,
+  onStateChange({ state, prevState }) {
+    if (state.query !== prevState.query) {
+      searchAndShowMainResults(state.query);
```

Comment:
We could better separate concerns:

```ts
const hitsContainer = document.querySelector('#hits');
// ...
index.search(query).then(result => render({ hits: result.hits }, hitsContainer));
```

Reply:
https://github.com/algolia/doc-code-samples/pull/232/commits/3f2db5658e62aa4c122db953937ed99112b651f2